galaxyproject/galaxy
# OIDC tokens

#    Refresh and expose OIDC tokens
    
The idea is to be able to use Galaxy login via an OIDC provider as a single sign-on and
 authenticate when needed during workflow execution using OIDC tokens. "When needed"
may include storage access, compute nodes access, etc. Additionally, we make sure 
tokens are always fresh, so for each API request, check if tokens are not expired and refresh otherwise.
If the refresh fails, the user will be logged out. 


## How to test the changes?
(Select all options that apply)
  1. Login using an OIDC provider (Keycloak, Azure AD, Google, etc.)
  2. Add expose a token to a tool, e.g.
  
        ```
           <environment_variables>
                <environment_variable name="_GALAXY_OIDC_ID_TOKEN" inject="oidc_id_token"/>
           </environment_variables>
        ``` 
  3. Check for `Refreshed user token...` message in log files (need to run Galaxy long enough for a token to expire).

## License
